modules/afsocket/afsocket-source.c

```diff
@@ -54,6 +55,33 @@ typedef struct _AFSocketSourceConnection
 
 static void afsocket_sd_close_connection(AFSocketSourceDriver *self, AFSocketSourceConnection *sc);
 
+static void
+_connections_count_set(AFSocketSourceDriver *self, gint value)
+{
+  self->num_connections = value;
+  stats_counter_set(self->num_connections_stats, value);
+}
+
+static gint
+_connections_count_get(AFSocketSourceDriver *self)
+{
+  return self->num_connections;
```

Comment:
I think we could easily use stats_counter_get() here and eliminate the member in self. Or am I missing something?

Reply:
I discussed it with multiple team members IRL:
- use two variables
- or use a stats variable for multiple purpose

More people preferred the later one. I am okey with both solution.

Reply:
and if we did, we could just eliminate these functions as all of them would only touch a single stats counter, that would probably be just fine at the call sites.

maybe I would extract the function (and only that) that decides if we can accept a new connection.

Reply:
I wouldn't build any logic in code on the exact values of stats counters.
The lifecycle/lifetime of stats clusters are different than drivers', the registration/constructor of these counters are also complex.

In this special case, we could guarantee a valid value, because the counter is set explicitly after a reload (depending on the `keep-alive()` option (`afsocket_sd_restore_kept_alive_connections`)), so this could work, but in general, I think it's not a good idea to make decisions in code based on the exact values of stats counters.

Reply:
I think if we have to improve the diagnostic capabilities of syslog-ng then we should have to provide a generic way for getting internal values without the aforementioned lifecycle problem, with a better solution than this.

I know that this work is a support related work,  so won't block the PR.